mita/openstack.py

```diff
@@ -157,47 +157,73 @@ def _get_flavor(self) -> NodeSize:
             logger.error(be)
             raise OpenStackDriverError("Encountered an unknown exception.")
 
-    def _has_free_ip_address(self, network: str) -> bool:
+    def _get_network_by_name(self, name: str) -> OpenStackNetwork:
         """
-        Return True if the given network has an IPv4 Address to lease.
+        Retrieve the OpenStackNetwork instance using the provided name.
+
+        Args:
+            name:   the name of the network.
+
+        Returns:
+            OpenStackNetwork instance
+
+        Raises:
+            ResourceNotFound when no matching resource was found.
+        """
+        url = f"{self.driver_v2._networks_url_prefix}?name={name}"
+        object_ = self.driver_v2.network_connection.request(url).object
+        networks = self.driver_v2._to_networks(object_)
+
+        if not networks:
+            raise ResourceNotFound(f"No network resource with name {name} found.")
+
+        return networks[0]
+
+    def _has_free_ip_address(self, net: OpenStackNetwork) -> bool:
+        """
+        Return True if the given network has an ip address to lease.
 
         Arguments:
-            network: String, The network UUID
+            net:    The reference to OpenStackNetwork
 
         Returns:
             bool - True on success else False
 
         Note: This method returns True only if the given network has more than 3
         """
         try:
-            resp = self.driver_v2.network_connection.request(
-                "/v2.0/network-ip-availabilities/{}".format(network),
+            url = f"/v2.0/network-ip-availabilities/{net.id}"
+            resp = self.driver_v2.network_connection.request(url)
+            subnets = resp.object.get("network_ip_availability", {}).get(
```

Comment:
I think this is good that you're being careful with the data by using `.get()`, however it makes it a little harder to tell if the code breaks later.

As one example, `network_ip_availability` is always going to be present there, right? (I'm asking because I don't know)

For the fields that we *know* will always be present, I would rather see us raise the `KeyError`s here immediately, because otherwise we could cruise on through this code without logging any errors and then find no `_free_ips` for anything.